pandas/tests/dtypes/test_cast.py

```diff
@@ -213,6 +213,13 @@ def test_maybe_convert_scalar(self):
         result = maybe_convert_scalar(Timedelta('1 day 1 min'))
         assert result == Timedelta('1 day 1 min').value
 
+    def test_maybe_infer_to_datetimelike(self):
+
+        # this construction raises IndexError in pandas=0.20.1
+        result = DataFrame(np.array([[NaT, 'a', 'b', 0],
```

Comment:
use both examples from here:  https://github.com/pandas-dev/pandas/issues/16362#issuecomment-301581166

and assert the result to an expected result == 8